lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -758,6 +758,42 @@ OpFoldResult AndOp::fold(ArrayRef<Attribute> constants) {
   return constFoldAssociativeOp(constants, hw::PEO::And);
 }
 
+/// returns a single common operand that all inputs of the operation `op` can
+/// be traced back to, or an empty `Value` if no such operand exists.
+///
+/// for example for `or(a[0], a[1], ..., a[n-1])` this function returns `a`
+/// (assuming the bit-width of `a` is `n`)
+template <typename Op>
+static Value getCommonOperand(Op op) {
+  if (!op.getType().isInteger(1))
+    return Value();
+
+  auto inputs = op.inputs();
+  size_t size = inputs.size();
+
+  auto sourceOp = inputs[0].template getDefiningOp<ExtractOp>();
+  if (!sourceOp)
+    return Value();
+  Value source = sourceOp.getOperand();
+
+  // fast path: the source bit-width does not match the size of the inputs
+  if (!source || size != source.getType().getIntOrFloatBitWidth())
+    return Value();
```

Comment:
I guess this condition is not necessary because (1)` source` is always nonnull and (2) `size != source.getType().getIntOrFloatBitWidth()` is correct only when `source` is 2bit integer. Probably you might want to just delete  this if-statement.